src/main/java/com/epam/aidial/cfg/domain/service/AdapterService.java

```diff
@@ -69,18 +75,34 @@ public void create(Adapter adapter) {
     @Transactional
     public void update(String adapterName, Adapter adapter) {
         adapterValidator.validateUpdate(adapterName, adapter);
-        AdapterEntity adapterEntity = adapterJpaRepository.findById(adapterName)
-                .orElseThrow(() -> new EntityNotFoundException(NOT_FOUND_MESSAGE_TEMPLATE.formatted(adapterName)));
+        AdapterEntity adapterEntity = findByAdapterName(adapterName);
         Optional.of(adapter)
                 .map(domainModel -> mapper.toEntity(domainModel, adapterEntity))
                 .map(adapterJpaRepository::save)
                 .orElseThrow(() -> new RuntimeException("Unable to update adapter " + adapter.getName()));
     }
 
     @Transactional
-    public void delete(String adapterName) {
-        assertExists(adapterName);
-        adapterJpaRepository.deleteById(adapterName);
+    public void delete(String adapterName, boolean removeModel) {
+        AdapterEntity adapterEntity = findByAdapterName(adapterName);
+        List<ModelEntity> models = adapterEntity.getModels();
+
+        if (CollectionUtils.isEmpty(models)) {
```

Comment:
adapter entity removing DRY. and early return.

more sustainable way is:
if (models is not empty) {
   remove models
 }
 
 remove adapter

Reply:
Done